packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1357,7 +1357,7 @@ class ComboBoxInternal extends React.Component<IComboBoxInternalProps, IComboBox
     const isChecked: boolean = this._isOptionChecked(item.index);
     const optionStyles = this._getCurrentOptionStyles(item);
     const optionClassNames = getComboBoxOptionClassNames(this._getCurrentOptionStyles(item));
-    const title = getPreviewText(item);
+    const title = item.title !== undefined ? item.title : getPreviewText(item);
```

Comment:
I think you could just do this here:

```suggestion
    const title = item.title ?? getPreviewText(item);
```